### ISSUE TITLE
[Bug]: Segmentation Fault on basic hypertable query

### ISSUE THREAD

**Issue by jgiacobbi** [timescale/timescaledb#7471]
### What type of bug is this?

Crash

### What subsystems and features are affected?

Other

### What happened?

I'm seeing a segmentation fault on a simple aggregate query of rows per timestamp: As you can see in the logs:
`select count(*), time from option_price group by time order by time desc limit 15;`

I tried a few permutations but they failed. Running the same query by time::date instead of time seems to work:
```
select count(*), time::date from option_price group by time::date order by time::date desc limit 2;
  count   |    time
----------+------------
 79559344 | 2024-11-11
  1388623 | 2024-11-08
  ```
  
Reducing the limit on the query by time to 2 doesn't change anything.

### TimescaleDB version affected

2.16.1

### PostgreSQL version used

16

### What operating system did you use?

Ubuntu 22.04 aarch64

### What installation method did you use?

Deb/Apt

### What platform did you run on?

On prem/Self-hosted

### Relevant log output and stack trace

```bash
2024-11-20 20:02:26.661 UTC [3165200] LOG:  server process (PID 3704184) was terminated by signal 11: Segmentation fault
2024-11-20 20:02:26.661 UTC [3165200] DETAIL:  Failed process was running: select count(*), time from option_price group by time order by time desc limit 15;
2024-11-20 20:02:26.661 UTC [3165200] LOG:  terminating any other active server processes
2024-11-20 20:02:26.662 UTC [3704188] postgres@intraday FATAL:  the database system is in recovery mode
```

### How can we reproduce the bug?

```bash
I'm at a loss for how to reproduce this exactly, so I'll take any advice on where to look. Notably it doesn't happen in our dev instance, which is not very busy. It does happen in our test instance where a lot of people are running investigative workloads. I don't think anyone is writing anything to that table or changing db structure, it should all be read activity. I wish I could fill out more detail here; any guidance would be helpful.
```

**Comment by fabriziomello** [comment]
Hey @jgiacobbi thanks for reaching out. Some questions:
1. can u provide a stack trace of the failure?
2. are u using compression on the hypertable?

**Comment by jgiacobbi** [comment]
Hello @fabriziomello . Working on getting a stack trace today. There is no compression policy on any of our hypertables because we had unrelated issues with compression.

A little update, I loaded 4 months of data on a smaller table into dev, and now dev is exhibiting the seg fault. That server is still mostly unused so I can rule out unknown third party bad behavior.

These queries work:
```
select max(last_price), security_id from security_price group by security_id;
select max(last_price), time from security_price group by time;
select count(*) from security_price;
select time from security_price group by time;
```

These queries seg fault:
```
select count(*), security_id from security_price group by security_id;
select count(*), time from security_price group by time order by time desc;
```

And like before, aggregating time by date first also works:
```
select count(*), time::date from security_price group by time::date;
```

Will update as soon as I have a stack trace.

**Comment by jgiacobbi** [comment]
Bit of a jam here. Went through the instructions here https://wiki.postgresql.org/wiki/Getting_a_stack_trace_of_a_running_PostgreSQL_backend_on_Linux/BSD

After I attached gdb to the process the crash stopped happening. Detaching gdb and restarting the service didn't change anything, the crash stopped. I truncated the table, reloaded 4 months of data, and no crash.

So I went to the test instance where it was still happening, ran postgres with `LimitCORE=infinity`, installed coredumpctl, got a useless coredump. Installed `postgresql-16-dbgsym` and the crash stopped happening. I removed dbgsym and the crash continues to not happen.

At this point I'm going to drop the whole database, run our IAC from scratch with and without debug support installed to try to get a useful stack trace, but if you have any advice I'm all ears.

**Comment by fabriziomello** [comment]
@jgiacobbi witch Postgres minor version are u using??? We faced issues with PG 16.5 because they introduced an [ABI breakage](https://www.postgresql.org/message-id/CABOikdNmVBC1LL6pY26dyxAS2f%2BgLZvTsNt%3D2XbcyG7WxXVBBQ%40mail.gmail.com). Then yday [they released new versions](https://www.postgresql.org/about/news/postgresql-172-166-1510-1415-1318-and-1222-released-2965/) fixing this problem.

So maybe when you install the postgresql-16-dbgsym it was updated to 16.6. So probably before you was running the buggy 16.5.
